core/drivers/crypto/se050/adaptors/apis/sss.c

```diff
@@ -0,0 +1,276 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <crypto/crypto.h>
+#include <kernel/panic.h>
+#include <se050.h>
+#include <se050_utils.h>
+#include <string.h>
+
+extern TEE_Result se050_core_early_init(struct se050_scp_key *keys);
+
+/*
+ * policies
+ */
+
+/* currently only asymmetric keys are stored in the SE050 NVM */
+static const sss_policy_u asym_key = {
+	.type = KPolicy_Asym_Key,
+	.auth_obj_id = 0,
+	.policy = {
+		.asymmkey = {
+			.can_Sign = 1,
+			.can_Verify = 1,
+			.can_Encrypt = 1,
+			.can_Decrypt = 1,
+			.can_KD = 1,
+			.can_Wrap = 1,
+			.can_Write = 1,
+			.can_Gen = 1,
+			.can_Import_Export = 1,
+			.can_KA = 1,
+			.can_Read = 1,
+			.can_Attest = 1,
+		}
+	}
+};
+
+static const sss_policy_u common = {
+	.type = KPolicy_Common,
+	.auth_obj_id = 0,
+	/* object can only be deleted by a secured and authenticated session */
+	.policy = {
+		.common = {
+			.can_Delete = 1,
+			.req_Sm = 1,
+		},
+	},
+};
+
+/* asym keys policy */
+sss_policy_t se050_asym_policy = {
+	.nPolicies = 2,
+	.policies = { &asym_key, &common },
+};
+
+sss_status_t se050_rotate_scp03_keys(struct sss_se05x_ctx *ctx)
+{
+	struct s050_scp_rotate_cmd cmd = { 0 };
```

Comment:
for all `struct xxx` initialization, prefer `{ }` over `{ 0 }`. Easier to maintain in case the structure 1st field change to something not a numerical value.

Reply:
ok